spec/20-http_header_format.md

```diff
@@ -102,14 +102,24 @@ trace-flags      = 2HEXDIGLC   ; 8 bit flags. Currently, only one bit is used. S
 
 This is the ID of the whole trace forest and is used to uniquely identify a [distributed trace](https://w3c.github.io/trace-context/#dfn-distributed-traces) through a system. It is represented as a 16-byte array, for example, `4bf92f3577b34da6a3ce929d0e0e4736`. All bytes as zero (`00000000000000000000000000000000`) is considered an invalid value.
 
-
 A vendor SHOULD generate globally unique values for `trace-id`. Many unique identification generation algorithms create IDs where one part of the value is constant (often time- or host-based), and the other part is a randomly generated value. Because tracing systems may make sampling decisions based on the value of `trace-id`, for increased interoperability vendors MUST keep the random part of `trace-id` ID on the left side.
 
-
-When a system operates with a `trace-id` that is shorter than 16 bytes, it SHOULD fill-in the extra bytes with random values rather than zeroes. Let's say the system works with an 8-byte `trace-id` like `3ce929d0e0e4736`. Instead of setting `trace-id` value to `0000000000000003ce929d0e0e4736` it SHOULD generate a value like `4bf92f3577b34da6a3ce929d0e0e4736` where `4bf92f3577b34da6a` is a random value or a function of time and host value.
-
-
-**Note**: Even though a system may operate with a shorter `trace-id` for [distributed trace](https://w3c.github.io/trace-context/#dfn-distributed-traces) reporting, the full `trace-id` MUST be propagated to conform to the specification.
+When a system operates with a `trace-id` that is shorter than 16 bytes, on new
+`trace-id` generation it SHOULD fill-in the extra bytes with random values
+rather than zeroes. Let's say the system works with an 8-byte `trace-id` like
+`23ce929d0e0e4736`. Instead of setting `trace-id` value to
+`23ce929d0e0e47360000000000000000` (note, that random part is kept on the left
```

Comment:
As mentioned by others, this example is backwards and doesn't match the comment in parenthesis.

Reply:
random part is kept to the left. Why is it not matching?

Reply:
I think the big problem here is the words "random part" can be interpreted multiple ways. The text says that the "random part is kept on the left" but I think it means to say "the original 8-byte trace ID is kept on the left". Yes/no? This section is discussing backfilling with random values vs. zeros, so when you say "random part is kept on the left" I'm reading that as the random backfilling on the left. But you're using the original 8-byte trace ID on the left, not random-backfill values, *and* the right side is all zeros, which nobody here is arguing should be done. 

And looking at it from the zeros point of view: why is the `0000000000000000` on the right here? The text says "Instead of setting `trace-id` value to
`23ce929d0e0e47360000000000000000`". Who would ever set it to `23ce929d0e0e47360000000000000000`? If it was zero-padded wouldn't it be `000000000000000023ce929d0e0e4736`? i.e. the padded zeros would be on the left.